components/batch-system/src/batch.rs

```diff
@@ -383,9 +416,13 @@ pub struct BatchSystem<N: Fsm, C: Fsm> {
     low_receiver: channel::Receiver<FsmTypes<N, C>>,
     pool_size: usize,
     max_batch_size: usize,
-    workers: Vec<JoinHandle<()>>,
+    workers: Arc<Mutex<Vec<JoinHandle<()>>>>,
     reschedule_duration: Duration,
     low_priority_pool_size: usize,
+    alive_workers: Arc<AtomicUsize>,
```

Comment:
> What `alive_workers` are used for?

@NingLin-P  It's used to record how many pollers there are currently, mainly used in conjunction with `expect_pool_size`.

eg: When we need to reduce the number of pollers from 8 to 4,  we immediately set expect_pool_size to 4,  after a while, when each `Poller` executes drop, it will reduce `alive_workers` by 1 and compare it with `expect_pool_size`. When it is equal, write `channel` to notify `config manager's dispatch` that our modification has been completed.